.github/workflows/ci.yml

```diff
@@ -39,11 +39,12 @@ jobs:
         submodules: recursive
 
     - name: Run with Docker
-      shell: 'script -q -e -c "bash {0}"'
-      env:
-        CONTAINER_NAME: ${{ matrix.container }}
+      shell: 'script -q -e -c "bash -xe {0}"'
       run: |
         chmod -R ugo+w .
+        if [ -n "${{ matrix.container }}" ] ; then
+          export CONTAINER_NAME="${{ matrix.container }}"
+        fi
         make -f misc/docker-ci/check.mk ${{ matrix.task }}
```

Comment:
How about changing this line to `${{ matrix.command }}` and defining the complete make command that has to be run for each flavor? I.e.,

```
        include:
          - type: default
            command: make -f misc/docker-ci/check.mk all
          - type: OpenSSL 1.1.0
            command: make -f misc/docker-ci/check.mk ossl1.1.0
          (snip)
          - type: DTrace
            command: make -f misc/docker-ci/check.mk dtrace CONTAINER=h2oserver/h2o-ci:ubuntu2004
```

This approach is in align with what we have been doing in travis.yml. In addition, I think it is more intuitive, as it show what it does instead of having a place for defining arguments and another for building a command based on them.

Reply:
Sounds good to me. It looks simpler than adding if-clause to the step script.